### PR TITLE
fix: updated filters query to exclude empty service users

### PR DESCRIPTION
**Pull request by Yakitrak**
## Describe your changes

Ticket: https://pins-ds.atlassian.net/browse/ASB-2430

Get representations by case reference endpoint /api/v1/representations?caseReference=BC0110001 is correctly filtering out representations which don't have a linked service user but doesn't correctly do this for the filter counts. This PR fixes that issue.

## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/applications-service-api/src/repositories/representation.backoffice.repository.js [hunk 1]
```diff
@@ -89,15 +88,20 @@ const getRepresentations = async (options) => {
...
 const getFilters = async (caseReference) => {
-	const sql = Prisma.sql`
-      SELECT DISTINCT(representationType), status, count(id) as total
-      FROM Representation
-      WHERE caseReference = ${caseReference}
-        AND (status = 'PUBLISHED' or status = 'published')
-        AND representationType is not null
-      GROUP BY representationType, status`;
+	const options = await prismaClient.representation.groupBy({
+		by: ['representationType'],
```
Comment (Philip-Philippou1) [inline packages/applications-service-api/src/repositories/representation.backoffice.repository.js:92]: Do we definitely want to omit the Group By status?

Comment (Yakitrak) [reply packages/applications-service-api/src/repositories/representation.backoffice.repository.js:92]: We weren't using status in the counts so I thought we did not need it - happy to add it though. Although as I am testing this, I am seeing issues with the count so need to make some changes.

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**